lib/Containers/MerkleTree.cpp

```diff
@@ -723,8 +719,7 @@ template<typename Hasher, std::uint64_t const BranchingBits>
 std::vector<std::pair<std::uint64_t, std::uint64_t>>
 MerkleTree<Hasher, BranchingBits>::diff(
     MerkleTree<Hasher, BranchingBits>& other) {
-  std::shared_lock<std::shared_mutex> guard1(_dataLock);
```

Comment:
Also, what do you think about using boost::upgrade_mutex and upgrade_lock here instead of shared_mutex?
In that case, we can't call diff parallel with diff, but can call other functions then diff working

Reply:
I don't think this is necessary since this code path is not that performance critical.